plugins/discovery-ec2/src/main/java/org/elasticsearch/plugin/discovery/ec2/Ec2DiscoveryPlugin.java

```diff
@@ -123,4 +139,46 @@ public void onModule(DiscoveryModule discoveryModule) {
         // Register cloud node settings: cloud.node
         AwsEc2Service.AUTO_ATTRIBUTE_SETTING);
     }
+
+    /** Adds a node attribute for the ec2 availability zone. */
+    @Override
+    public Settings additionalSettings() {
+        return getAvailabilityZoneNodeAttributes(settings, AwsEc2ServiceImpl.EC2_METADATA_URL + "placement/availability-zone");
+    }
+
+    // pkg private for testing
+    static Settings getAvailabilityZoneNodeAttributes(Settings settings, String azMetadataUrl) {
+        if (AwsEc2Service.AUTO_ATTRIBUTE_SETTING.get(settings) == false) {
+            return Settings.EMPTY;
+        }
+        Settings.Builder attrs = Settings.builder();
+
+        final URL url;
+        final URLConnection urlConnection;
```

Comment:
I don't think we can since I don't think you can bind to the special ip? aws does networking tricks to make that metadata lookup work.